test/boost/json_test.cc

```diff
@@ -54,3 +68,163 @@ BOOST_AUTO_TEST_CASE(test_parsing_map_from_null) {
     BOOST_REQUIRE(map1 == map2);
     BOOST_REQUIRE(map1 == empty_map);
 }
+
+namespace {
+
+std::string_view trim(std::string_view sv) {
+    auto it = sv.begin();
+    auto end = sv.end();
+    while (it != end && *it == ' ') {
+        ++it;
+    }
+    return std::string_view(it, end);
+}
+
+std::string_view cmp_and_remove_prefix(std::string_view sv, std::string_view expected) {
+    BOOST_TEST_MESSAGE(format("cmp_and_remove_prefix: {} expected='{}'", sv, expected));
+    trim(sv);
+    BOOST_REQUIRE(sv.starts_with(expected));
+    auto sz = expected.size();
+    sv = sv.substr(sz, sv.size() - sz);
+    return trim(sv);
+}
+
+void verify_parenthesis(std::string_view sv) {
+    static std::unordered_map<char, char> paren_map = {{'{', '}'}, {'[', ']'}, {'(', ')'}, {'<', '>'}};
+
+    char open = sv[0];
+    char close = sv[sv.size() - 1];
+    auto it = paren_map.find(open);
+    if (it == paren_map.end()) {
+        BOOST_FAIL(format("Unexpected delimiters: '{}' '{}'", open, close));
+    }
```

Comment:
nit,
```c++
BOOST_REQUIRE_MESSAGE(
    it != paren_map.end(), 
    format("Unexpected delimiters: '{}' '{}'", open, close));
```

Reply:
sorry, I think it's less intuitive.
I don't see the value in this macro.